be1-go/hub/standard_hub/message_handling.go

```diff
@@ -269,6 +269,17 @@ func (h *Hub) handlePublish(socket socket.Socket, byteMessage []byte) (int, erro
 		return -1, xerrors.Errorf("failed to unmarshal publish message: %v", err)
 	}
 
+	signature := publish.Params.Message.Signature
+	messageId := publish.Params.Message.MessageID
+	data := publish.Params.Message.Data
+
+	expectedMessageId := messagedata.Hash(data, signature)
+
+	if expectedMessageId != messageId {
+
+		return publish.ID, xerrors.Errorf("message_id is wrong expected is %q but was %v", expectedMessageId, messageId)
```

Comment:
I may be missing something but would it not be massively vulnerable to inform the sender of the hash that we expected ? Could he not resend the same message with the appropriate message_id thus impersonating someone else ?

Reply:
I do agree, the attacker could just send a random message id to get information about the true message id and would just resend the same message with the correct message id.